### ISSUE TITLE
Can the Keras CTC_Loss be negative?

### ISSUE THREAD

**Issue by aayushee** [keras-team/keras#9369]
Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps

- [ ] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by aayushee** [comment]
I am using CTC Loss from Keras API as posted in the image OCR example to perform online handwritten recognition with a 2-layer Bidirectional LSTM model. But I am getting negative loss after training for 7 epochs. Can this loss value be negative? 
And how do I know when the network has trained well (how do I set number of epochs)?
I have attached the model output here:
![image](https://user-images.githubusercontent.com/7151691/36092156-602170e4-100c-11e8-949a-713955676850.png)

**Comment by theceday** [comment]
I am running image_ocr with tensorflow 1.6 gpu backend in windows.
Only change i made to code is disabling multi_fonts, let it run only with Courrier.

Here is log of last two epochs ( though i have seen negative values in earlier epochs)

[logs](https://gist.github.com/theceday/8e6c27bf6b56754193ac46e99f70ef3a)

**Comment by aayushee** [comment]
Hi, Thanks for replying.
I understand that the loss value can come out to be negative as you mentioned. But how do you know when to stop training in this case? How do you decide on the number of epochs?

**Comment by northeastsquare** [comment]
@aayushee Hi, I also met this problem: negtive ctc loss, and then nan.
Have you solved it?
